ui/webui/src/components/storage/StorageConfiguration.jsx

```diff
@@ -0,0 +1,350 @@
+                        selectedScenarioId = scenario.id;
+                    }
+                }
+            }
+            setSelectedScenario(selectedScenarioId);
+            setIsFormValid(availableScenarioExists);
+        };
+
+        updateScenarioState(scenarios);
+    }, [scenarios, setIsFormValid]);
+
+    useEffect(() => {
+        const applyScenario = async (scenarioId) => {
+            const scenario = scenarios.filter(s => s.id === scenarioId)[0];
+            console.log("Updating scenario selected in backend to", scenario.id);
+            await setInitializationMode({ mode: scenario.initializationMode }).catch(console.error);
+        };
+        if (selectedScenario) {
+            applyScenario(selectedScenario);
+        }
+    }, [scenarios, selectedScenario]);
+
+    const updateDetailContent = (scenarioId) => {
+        const scenario = scenarios.filter(s => s.id === scenarioId)[0];
+        const hint = scenarioAvailability[scenarioId].hint;
+        setDetailContent(scenarioDetailContent(scenario, hint));
+    };
+
+    const onScenarioToggled = (scenarioId) => {
+        setSelectedScenario(scenarioId);
+        updateDetailContent(scenarioId);
+    };
+
+    const showScenarioDetails = (scenarioId) => {
+        updateDetailContent(scenarioId);
+        setIsDetailExpanded(!isDetailExpanded);
+    };
+
+    const scenarioItems = scenarios.map(scenario =>
+        <DataListItem key={scenario.id}>
+            <DataListItemRow>
+                <DataListItemCells dataListCells={[
+                    <DataListAction key="radio">
+                        {!scenarioAvailability[scenario.id].available &&
+                        <Tooltip
+                          aria-live="polite"
+                          content={scenarioAvailability[scenario.id].shortHint}
+                          reference={() => document.getElementById(idPrefix + "-autopart-scenario-" + scenario.id)}
+                        />}
+                        <Radio
+                          id={idPrefix + "-autopart-scenario-" + scenario.id}
+                          value={scenario.id}
+                          name="autopart-scenario"
+                          label={scenario.label}
+                          isDisabled={!scenarioAvailability[scenario.id].available}
+                          isChecked={selectedScenario === scenario.id}
+                          onChange={() => onScenarioToggled(scenario.id)}
+                        />
+                    </DataListAction>,
+                    <DataListCell key="more">
+                        {scenarioAvailability[scenario.id].reason &&
+                        <Flex spaceItems={{ default: "spaceItems2xl" }}>
+                            <FlexItem />
+                            <FlexItem>
+                                <HelperText>
+                                    <HelperTextItem variant="warning" icon=<ExclamationTriangleIcon />>
```

Comment:
it's generally more readable to put the <ExclamationTriangleIcon /> inside curly brackets {}.

Reply:
Ah, you mean `icon={<ExclamationTriangleIcon />}` I guess, I'll use the brackets.